From cf85454d710e03abc074645585fa34b9a9a9e240 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Mon, 30 May 2016 03:56:38 +0100
Subject: [PATCH 10/12] pgp-verbose-mime

Configurable PGP mime part name path by Vineet Kumar

Orignal patch from http://www.doorstop.net/projects
---
 PATCHES   |  1 +
 globals.h |  2 ++
 init.h    | 12 ++++++++++++
 pgp.c     |  3 ++-
 4 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/PATCHES b/PATCHES
index 9db458d..35daff7 100644
--- a/PATCHES
+++ b/PATCHES
@@ -1,3 +1,4 @@
+patch-1.5.11.vk.pgp_verbose_mime
 dgc.subjrx
 fg.change_folder_next
 vvv.quote
diff --git a/globals.h b/globals.h
index 0a27720..8451bcd 100644
--- a/globals.h
+++ b/globals.h
@@ -273,6 +273,8 @@ WHERE REGEXP PgpDecryptionOkay;
 WHERE char *PgpSignAs;
 WHERE short PgpTimeout;
 WHERE char *PgpEntryFormat;
+WHERE char *PgpMimeSignatureFilename;
+WHERE char *PgpMimeSignatureDescription;
 WHERE char *PgpClearSignCommand;
 WHERE char *PgpDecodeCommand;
 WHERE char *PgpVerifyCommand;
diff --git a/init.h b/init.h
index 2c260d7..1b2bfe6 100644
--- a/init.h
+++ b/init.h
@@ -3222,6 +3222,18 @@ struct option_t MuttVars[] = {
   ** a line quoted text if it also matches $$smileys. This mostly
   ** happens at the beginning of a line.
   */
+  { "pgp_mime_signature_filename", DT_STR, R_NONE, UL &PgpMimeSignatureFilename, UL "signature.asc"},
+  /*
+  ** .pp
+  ** This option sets the filename used for signature parts in PGP/MIME
+  ** signed messages.
+  */
+  { "pgp_mime_signature_description", DT_STR, R_NONE, UL &PgpMimeSignatureDescription, UL "Digital signature"},
+  /*
+  ** .pp
+  ** This option sets the Content-Description used for signature parts in
+  ** PGP/MIME signed messages.
+  */
 
 
 
diff --git a/pgp.c b/pgp.c
index 6a6f3c9..6265a7f 100644
--- a/pgp.c
+++ b/pgp.c
@@ -1261,7 +1261,8 @@ BODY *pgp_sign_message (BODY *a)
   t->disposition = DISPNONE;
   t->encoding = ENC7BIT;
   t->unlink = 1; /* ok to remove this file after sending. */
-  mutt_set_parameter ("name", "signature.asc", &t->parameter);
+  mutt_set_parameter ("name", PgpMimeSignatureFilename, &t->parameter);
+  t->description = safe_strdup (PgpMimeSignatureDescription);
 
   return (a);
 }
-- 
2.9.0

